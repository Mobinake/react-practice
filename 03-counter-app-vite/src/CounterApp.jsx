import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);

	const handleAdd = () => setCounter((c) => c + 1);

	const handleDecrease = () => setCounter((c) => c - 1);

	const handleReset = () => setCounter(() => value);

	return (
		<>
			<h1>CounterApp</h1>
			<h2 data-testid={'counter'}> {counter} </h2>
			<button
				aria-label='btn-add'
				onClick={handleAdd}>
				+1
			</button>
			<button
				aria-label='btn-decrease'
				onClick={handleDecrease}>
				-1
			</button>
			<button
				aria-label='btn-reset'
				onClick={handleReset}>
				Reset
			</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};
