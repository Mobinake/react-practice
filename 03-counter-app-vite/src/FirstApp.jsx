import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle }) => {
	return (
		<>
			<h1 data-testid='test-title'>{title}</h1>
			<p>{subTitle}</p>
			<p>Prueba de comp</p>
		</>
	);
};

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
	title: 'No hay t√≠tulo',
	subTitle: 100,
};
