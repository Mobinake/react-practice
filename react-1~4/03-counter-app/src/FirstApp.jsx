import PropTypes from 'prop-types';

const newMessage = {
	message: 'Hola, react.',
	title: 'Mobin',
};

export function FirstApp({ title, subtitle }) {
	return (
		<>
			{/* se puede colocar cualquier codigo de JS aqui, excepto funciones.
                aqui lo que hace es convertir la respuesta en JSON, para que no se devuelva como funcion*/}
			<code>{JSON.stringify(newMessage)}</code>
			<h1 data-testid='test-title'>{title}</h1>
			<h2>{subtitle}</h2>
		</>
	);
}

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
	title: 'No hay titulo',
	subtitle: 'No hay subtitulo',
};
