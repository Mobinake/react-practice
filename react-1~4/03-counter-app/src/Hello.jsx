export function Hello() {
	return <h1>Hola mamasita </h1>;
}
