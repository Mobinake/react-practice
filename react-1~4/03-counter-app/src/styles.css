html,
body {
	background-color: #21232a;
	color: #ffffff;
	font-family: Helvetica, Arial, sans-serif;
	font-size: 16px;
	padding: 70px;
}

button {
	background-color: #2fa3f0;
	border: none;
	color: black;
	padding: 8px 16px;
	text-align: center;
	font-size: 16px;
	border-radius: 16px;
}
