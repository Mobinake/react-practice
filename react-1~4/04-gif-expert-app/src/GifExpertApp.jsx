import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
	//categorias que se muestran
	const [categories, setCategories] = useState(['Vinland Saga']);

	const onAddCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;
		// console.log(newCategory);
		setCategories([newCategory, ...categories]);
	};

	return (
		<>
			{/* titulo */}
			<h1>GifExpertApp</h1>

			{/* input */}
			<AddCategory onNewCategory={onAddCategory} />

			{/* listado de gif */}
			{categories.map((category) => (
				<GifGrid
					key={category}
					category={category}
				/>
			))}
		</>
	);
};
