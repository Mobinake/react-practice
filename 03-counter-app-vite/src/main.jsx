// import { HelloWorld } from "./HelloWorld";
// import { FirstApp } from "./FirstApp";
import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		{/* <FirstApp title={"Hola"} subTitle={1234} /> */}
		<CounterApp value={0} />
	</React.StrictMode>
);
