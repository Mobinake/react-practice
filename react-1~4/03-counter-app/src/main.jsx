import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';

import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		<CounterApp value={1} />
		{/* <FirstApp title="Hola, soy Goku" subtitle="Soy un subtÃ­tulo" /> */}
	</React.StrictMode>
);
