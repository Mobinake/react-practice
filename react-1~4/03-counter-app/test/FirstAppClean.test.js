/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en FirstApp', () => {
	const title = 'Hola, soy Mobin';
	const subtitle = 'Ingeniero';

	test('debe hacer match con el snapshot', () => {
		const { container } = render(
			<FirstApp
				title={title}
				subtitle={subtitle}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	test('mostrar mensaje "Hola, soy Mobin"', () => {
		render(
			<FirstApp
				title={title}
				subtitle={subtitle}
			/>
		);
		expect(screen.getByText(title)).toBeTruthy();
	});

	test('debe mostrar title en h1', () => {
		render(
			<FirstApp
				title={title}
				subtitle={subtitle}
			/>
		);
		expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
			title
		);
	});

	test('debe mostrar subtitulo enviado por props', () => {
		render(
			<FirstApp
				title={title}
				subtitle={subtitle}
			/>
		);
		expect(screen.getAllByText(subtitle).length).toBe(1);
	});
});
