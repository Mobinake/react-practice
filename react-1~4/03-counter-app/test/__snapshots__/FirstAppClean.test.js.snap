// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <code>
    {"message":"Hola, react.","title":"Mobin"}
  </code>
  <h1
    data-testid="test-title"
  >
    Hola, soy Mobin
  </h1>
  <h2>
    Ingeniero
  </h2>
</div>
`;
